### PR TITLE
Ignore locales; fix kebab-cased translation keys

### LINKED ISSUE #2432: Issues with translation `json`s after #2271
## Description

There are two problems with the translation `json`s after #2271 was merged:
1. The CI fails because some languages use the `nbsp` (non-breaking space) character, and it fails the linting step in the CI.
2. Some translations have still have the old kebab-cased keys that look like "###read-more###" in the GlotPress interface. We cannot ensure that the translators change the keys manually to camelCase, so we should add a step to camelize keys locally when saving jsons, instead.

## Reproduction

The first problem was reported by @zackkrida in #2398:

> While trying to push this I also noticed that after the merging of #2271, some of the generated translation files fail eslint:

```shell
openverse/frontend/src/locales/de.json
  171:29  error  Irregular whitespace not allowed  no-irregular-whitespace
  236:29  error  Irregular whitespace not allowed  no-irregular-whitespace
  472:29  error  Irregular whitespace not allowed  no-irregular-whitespace
  706:29  error  Irregular whitespace not allowed  no-irregular-whitespace

openverse/frontend/src/locales/fr.json
   68:28   error  Irregular whitespace not allowed  no-irregular-whitespace
   68:108  error  Irregular whitespace not allowed  no-irregular-whitespace
   68:112  error  Irregular whitespace not allowed  no-irregular-whitespace
   68:135  error  Irregular whitespace not allowed  no-irregular-whitespace
   68:143  error  Irregular whitespace not allowed  no-irregular-whitespace
   68:148  error  Irregular whitespace not allowed  no-irregular-whitespace
   68:156  error  Irregular whitespace not allowed  no-irregular-whitespace
   68:161  error  Irregular whitespace not allowed  no-irregular-whitespace
   68:173  error  Irregular whitespace not allowed  no-irregular-whitespace
   71:32   error  Irregular whitespace not allowed  no-irregular-whitespace
   92:39   error  Irregular whitespace not allowed  no-irregular-whitespace
  182:48   error  Irregular whitespace not allowed  no-irregular-whitespace
  182:56   error  Irregular whitespace not allowed  no-irregular-whitespace

openverse/frontend/src/locales/lv.json
  11:24  error  Irregular whitespace not allowed  no-irregular-whitespace

openverse/frontend/src/locales/ru.json
  321:230  error  Irregular whitespace not allowed  no-irregular-whitespace
```

> I don't think we want to lint those files at all, so I've added them to the root .eslintignore. Does that seem right @obulat?

To see the second problem, you can run `just frontend/run i18n` and check one of the "<locale>.json" files. If you look at `nl.json`, and search for string "cc-search" or "read-more", you'll see that the key wasn't renamed to "ccSearch" or "readMore". This is why in the migration banner in `nl` locale (https://openverse.org/nl?referrer=creativecommons.org), you won't see "CC Search", or "read more" link.

## Additional context

I've marked this issue as critical because the PRs are failing CI without it.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #2432 by @obulat

## Description

This PR adds the locale `json` files to eslintignore as they are gitignored anyways. This CI fix was applied in #2398, but it's better to merge it ASAP because the CI fails without it.

Another fix is converting the keys (that are between `###` characters in `pot` files in GlotPress, and between `{}` brackets in Vue i18n json files) from kebab-case to camelCase.

## Testing Instructions

Run `just frontend/run i18n` to load the translation files. Then, go to `nl` locale with a migration notice: `http://localhost:8443/nl?referrer=creativecommons.org`. You should see the "read more" link and "CC Search" in the notice.
The CI eslint step shouldn't show the messages from the linked issue.

## Checklist

- [ ] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [ ] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [ ] My commit messages follow [best practices][best_practices].
- [ ] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/src/locales/scripts/utils.js [hunk 1]
```diff
@@ -34,7 +36,11 @@ const replacePlaceholders = (json) => {
     return null
   }
   if (typeof json === "string") {
-    return json.replace(/###([a-zA-Z-]*)###/g, "{$1}")
+    // Replace old kebab-cased keys that can still be in the translations with camelCased keys the app expects
```
Comment (zackkrida) [inline frontend/src/locales/scripts/utils.js:39]:
```suggestion
    // Replace old kebab-cased keys that can still be in the translations with camelCased keys the app expects
    // TODO: Remove this once all translation strings are updated.
```

**Review by zackkrida — APPROVED** [review]
👍 Great find.

**Review by sarayourfriend — APPROVED** [review]
LGTM.

What an interesting way to prevent the loss of those translations in the meantime. Do we have a set date for when the TODO would happen? Can we make it a blocked issue so MSR can check back in on it by a specific date?

**Comment by obulat** [comment]
> What an interesting way to prevent the loss of those translations in the meantime. Do we have a set date for when the TODO would happen? Can we make it a blocked issue so MSR can check back in on it by a specific date?

I'll add a warning to the replacer function if it finds a dash in the key so that there's a way to track this. Then, MSR could check if the warning is still there. 
I tried updating the keys manually in GlotPress, but because we have so many locales, it's pretty tedious, so it might take a long time.

**Pull request merged**